src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java

```diff
@@ -715,9 +728,28 @@ public static AcLoadFlowParameters createAcParameters(Network network, LoadFlowP
         return acParameters;
     }
 
+    private static NewtonRaphsonStoppingCriteria generateNewtonRaphsonStoppingCriteria(OpenLoadFlowParameters parametersExt,
+                                                                                       NewtonRaphsonStoppingCriteriaType typeCriteria) {
+        switch (typeCriteria) {
+            case DEFAULT_CRITERIA:
+                return new DefaultNewtonRaphsonStoppingCriteria(parametersExt.getNewtonRaphsonConvEpsPerEq());
+            case CUSTOM_CRITERIA:
+                return new CustomNewtonRaphsonStoppingCriteria(parametersExt.getNewtonRaphsonConvEpsPerEq(),
+                        parametersExt.maxActivePowerMismatch, parametersExt.maxReactivePowerMismatch, parametersExt.maxVoltageMismatch);
+            default:
+                return null;
```

Comment:
Throw an exception here rather than returning null. You can look at other switches to see what kind of exception you can throw in similar cases.